### DISCUSSION TITLE
help

### DISCUSSION THREAD

**Discussion by THEkatinamicrowave** [KadeArchive/Kade-Engine#2230] [Q&A]
I need each character in my mod to have their own Blueballed screen. For example, if the player character is Pico, it plays (and I don't have them added in yet) picoDeathScreen.png/.xml

How do I do that with Kade Engine

**Comment by THEkatinamicrowave** [comment]
Edit: Here is an example
![Worriedbob](https://user-images.githubusercontent.com/89645987/132961988-eb451514-c1e4-47fb-867d-bb11cb93d286.png)

**Reply by libracrumbs** [reply]
Just add the character in, it should work, if it does not, i dont know

**Reply by libracrumbs** [reply]
Btw use the code of " case 'bf' " or it wont work

**Reply by libracrumbs** [reply]
copy this 

case 'bf':

var tex = Paths.getSparrowAtlas('BOYFRIEND','shared',true);
				frames = tex;

				trace(tex.frames.length);

				animation.addByPrefix('idle', 'BF idle dance', 24, false);
				animation.addByPrefix('singUP', 'BF NOTE UP0', 24, false);
				animation.addByPrefix('singLEFT', 'BF NOTE LEFT0', 24, false);
				animation.addByPrefix('singRIGHT', 'BF NOTE RIGHT0', 24, false);
				animation.addByPrefix('singDOWN', 'BF NOTE DOWN0', 24, false);
				animation.addByPrefix('singUPmiss', 'BF NOTE UP MISS', 24, false);
				animation.addByPrefix('singLEFTmiss', 'BF NOTE LEFT MISS', 24, false);
				animation.addByPrefix('singRIGHTmiss', 'BF NOTE RIGHT MISS', 24, false);
				animation.addByPrefix('singDOWNmiss', 'BF NOTE DOWN MISS', 24, false);
				animation.addByPrefix('hey', 'BF HEY', 24, false);

				animation.addByPrefix('firstDeath', "BF dies", 24, false);
				animation.addByPrefix('deathLoop', "BF Dead Loop", 24, false);
				animation.addByPrefix('deathConfirm', "BF Dead confirm", 24, false);

				animation.addByPrefix('scared', 'BF idle shaking', 24);

				loadOffsetFile(curCharacter);

				playAnim('idle');

				flipX = true;

and edit it to your needs using the .xml file!

**Reply by libracrumbs** [reply]
For example:

case 'pico-playable':
				var tex = Paths.getSparrowAtlas('Pico_FNF_Assetss','shared',true);
				frames = tex;

				trace(tex.frames.length);

				animation.addByPrefix('idle', 'Pico Idle Dance', 24, false);
				animation.addByPrefix('singUP', 'pico Up note', 24, false);
				animation.addByPrefix('singLEFT', 'Pico NOTE LEFT', 24, false);
				animation.addByPrefix('singRIGHT', 'Pico Note Right', 24, false);
				animation.addByPrefix('singDOWN', 'Pico Down Note', 24, false);
				animation.addByPrefix('singUPmiss', 'pico Up note miss', 24, false);
				animation.addByPrefix('singLEFTmiss', 'Pico NOTE LEFT miss', 24, false);
				animation.addByPrefix('singRIGHTmiss', 'Pico Note Right Miss', 24, false);
				animation.addByPrefix('singDOWNmiss', 'Pico Down Note MISS', 24, false);
				

				animation.addByPrefix('firstDeath', "BF dies", 24, false); // This part is not in the pico xml file.
				animation.addByPrefix('deathLoop', "BF Dead Loop", 24, false); // This part is not in the pico xml file
				animation.addByPrefix('deathConfirm', "BF Dead confirm", 24, false); // this part is not in the pico xml file

				

				loadOffsetFile(curCharacter);

				playAnim('idle');

				flipX = true;

**Reply by libracrumbs** [reply]
Or, with some tweaks, you can use the original code of Pico just fine (but without the blueball animation, you will need to add that in by yourself.)

case 'pico':
				tex = Paths.getSparrowAtlas('Pico_FNF_assetss','shared',true);
				frames = tex;
				animation.addByPrefix('idle', "Pico Idle Dance", 24);
				animation.addByPrefix('singUP', 'pico Up note0', 24, false);
				animation.addByPrefix('singDOWN', 'Pico Down Note0', 24, false);
				if (isPlayer)
				{
					animation.addByPrefix('singLEFT', 'Pico NOTE LEFT0', 24, false);
					animation.addByPrefix('singRIGHT', 'Pico Note Right0', 24, false);
					animation.addByPrefix('singRIGHTmiss', 'Pico Note Right Miss', 24, false);
					animation.addByPrefix('singLEFTmiss', 'Pico NOTE LEFT miss', 24, false);
				}
				else
				{
					// Need to be flipped! REDO THIS LATER!
					animation.addByPrefix('singLEFT', 'Pico Note Right0', 24, false);
					animation.addByPrefix('singRIGHT', 'Pico NOTE LEFT0', 24, false);
					animation.addByPrefix('singRIGHTmiss', 'Pico NOTE LEFT miss', 24, false);
					animation.addByPrefix('singLEFTmiss', 'Pico Note Right Miss', 24, false);
				}

				animation.addByPrefix('singUPmiss', 'pico Up note miss', 24);
				animation.addByPrefix('singDOWNmiss', 'Pico Down Note MISS', 24);

				loadOffsetFile(curCharacter);

				playAnim('idle');

				flipX = true;

**Reply by libracrumbs** [reply]
Forgot to say, you don't need to make it a different spritesheet, you can use the same spritesheet.

**Reply by THEkatinamicrowave** [reply]
ok thanks I'll try it

**Comment by Burger-Gaming** [comment]
Go to GameOverSubstate.hx, find daBf and add a case for your bf character

**Reply by THEkatinamicrowave** [reply]
Ok I'll try
